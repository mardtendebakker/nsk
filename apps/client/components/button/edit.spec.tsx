import { fireEvent, render } from '@testing-library/react';
import Edit from './edit';

describe('edit', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Edit onClick={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('handles click', () => {
    const onClick = jest.fn();
    const { getByRole } = render(<Edit onClick={onClick} />);
    fireEvent.click(getByRole('button'));

    expect(onClick).toHaveBeenCalled();
  });
  it('disables click', () => {
    const onClick = jest.fn();
    const { getByRole } = render(<Edit onClick={onClick} disabled />);
    fireEvent.click(getByRole('button'));

    expect(onClick).not.toHaveBeenCalled();
  });
});
