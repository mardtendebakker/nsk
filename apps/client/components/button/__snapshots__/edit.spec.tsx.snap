// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit matches snapshot 1`] = `
<DocumentFragment>
  <span
    class=""
    data-mui-internal-clone-element="true"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1rb4sk0-MuiButtonBase-root-MuiIconButton-root"
      data-testid="edit-icon-button"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lr4af1-MuiSvgIcon-root"
        data-testid="EditIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </span>
</DocumentFragment>
`;
